firebase-firestore/src/main/java/com/google/firebase/firestore/UserDataConverter.java

```diff
@@ -207,7 +223,8 @@ private FieldValue convertAndParseFieldData(Object input, ParseContext context)
    * conform to document data (i.e. it must parse into an ObjectValue model type) and will throw an
    * appropriate error otherwise.
    */
-  private ObjectValue convertAndParseDocumentData(Object input, ParseContext context) {
+  private ObjectValue convertAndParseDocumentData(
+      Object input, ParseContext context, String expectedId) {
```

Comment:
Per Gil's feedback elsewhere, this probably needs to be a DocumentReference but if not, I'd recommend renaming `expectedId` to `expectedDocumentId` or similar, since `id` is probably too ambiguous by itself.

Reply:
No longer needed